### ISSUE TITLE
Allow magic methods __serialize and __unserialize in CamelCapsFunctionNameSniff

### ISSUE THREAD

**Issue by ludekbenedik** [squizlabs/PHP_CodeSniffer#3029]
**Describe the bug**
The sniff `Generic.NamingConventions.CamelCapsFunctionName.MethodDoubleUnderscore` does not known magic methods `__serialize` and `__unserialize`.

**Code sample**
```php
<?php declare(strict_types = 1);

namespace App;

class Test
{
    public function __serialize(): array
    {
        return [];
    }

    public function __unserialize(array $data): void
    {
    }
}
```

**To reproduce**
```
  7 | ERROR | Method name "Test::__serialize" is invalid; only PHP
    |       | magic methods should be prefixed with a double
    |       | underscore
    |       | (Generic.NamingConventions.CamelCapsFunctionName.MethodDoubleUnderscore)
 12 | ERROR | Method name "Test::__unserialize" is invalid; only PHP
    |       | magic methods should be prefixed with a double
    |       | underscore
    |       | (Generic.NamingConventions.CamelCapsFunctionName.MethodDoubleUnderscore)
```

**Expected behavior**
None error

**Versions (please complete the following information):**
 - OS: [*]
 - PHP: [7.4]
 - PHPCS: [3.5.5 (stable)]
 - Standard: [Generic]
